### DISCUSSION TITLE
Deleting Attribute Set products deleted

### DISCUSSION THREAD

**Discussion by TangLiang** [OpenMage/magento-lts#4364] [Ideas]
### Steps to reproduce (*)
1.Make a new attribute set
2.Assign attribute set to product
3.delete attribute set

### Actual result (*)
The product gets deleted from the DB and everywhere.

**Comment by justinbeaty** [comment]
The admin panel should give a warning about this, no? 

https://github.com/OpenMage/magento-lts/blob/7806f0e529da17a85df6f82c09420116a1a60f26/app/code/core/Mage/Adminhtml/Block/Catalog/Product/Attribute/Set/Main.php#L95-L105

Would it be better to not allow deletion of the attribute set if products are assigned to it? Or require the user to type something like “confirm deletion” into a box to proceed? Or maybe disable the affected products?

I believe it’s this way because I’m not sure reassigning the product to the default attribute set is possible since there could be required fields present in the default set that weren’t in the deleted set.

**Comment by TangLiang** [comment]
i think it's much better to reassign the product to default with basic attributes

**Comment by sreichel** [comment]
> The admin panel should give a warning about this, no?

Yes. But ... its stupid to delete products base on attribute-set.

Better add code from https://github.com/flagbit/Magento-ChangeAttributeSet as forked module in OM namespace?

**Comment by justinbeaty** [comment]
@sreichel Yes that would be nice, but that module is GPL... Can we get all contributors of that module to agree to relicense under OSL?

**Comment by sreichel** [comment]
GPL2 (not)3 shoulb be okay (???). I dont know if it makes a difference, but most of current code is written/changed by me.

Ping @hackwell.

**Comment by justinbeaty** [comment]
> GPL2 (not)3 shoulb be okay (???). I dont know if it makes a difference, but most of current code is written/changed by me.

I guess that if it stays as a separate repo it's fine as GPL, but to integrate into core I don't think GPLv2 is compatible either. There are not many contributors to the repo it seems and some of the old contributions might have been replaced by new code by now. So I don't think it would be too hard to relicense if everyone is willing, or to use only some parts of the code (i.e. parts you wrote) and replace others.

**Comment by addison74** [comment]
What is under discussion is the logic of the action. Do we delete all the products that use that attribute set or do we keep them? I guess this has been analyzed in the past by those who implemented it.

If someone wants to delete an attribute set then the associated products are also deleted, as out of the box Magento does not offer changing the attribute set for a product. Flagbit is a must have and I use it in all installations because for speed the products are first introduced on a default set then the set is changed when it is finally agreed. However, I would not agree to it being introduced into the OpenMage distribution because it can be a source of problems as its use requires a good understanding of what is happening behind the scenes.

On the one hand, it must be taken into account that setting up another set could leave orphaned records behind in the database. On the other hand, the products cannot remain as long as the attribute set has been deleted, I wouldn't see the point of having a temporary set. 

From experience, so far I have never deleted an attribute set in production.
